app/javascript/packs/contentEditable.js

```diff
@@ -1,14 +1,27 @@
-import UpdateQuery from './updateQuery.js'
+import TextEditableStrategy from './textEditableStrategy.js'
+import ImageEditableStrategy from './imageEditableStrategy.js'
 class ContentEditable {
-  constructor(selector, eventType, callback){
-    this.editables =  document.querySelectorAll(selector);
-    this.eventType = eventType;
-    this.callback = callback
+  constructor(editableKind){
+    this.editables =  document.querySelectorAll(`[data-editablekind='${editableKind}']`);
+    this.editionStrategy = this.getStrategy(editableKind)
   }
 
   init(){
     for (const element of this.editables){
-      element.addEventListener(this.eventType, () => this.callback(element))
+      new this.editionStrategy(element).apply();
+    }
+  }
+
+  getStrategy(editableKind){
+    switch(editableKind) {
+      case 'text':
+        return TextEditableStrategy
+        break;
+      case 'image':
+        return ImageEditableStrategy
+        break
+      default:
+        return TextEditableStrategy
```

Comment:
More scalable approach
```suggestion
    const strategiesMap = {
      'text': TextEditableStrategy,
      'image': ImageEditableStrategy,
    }
        return strategiesMap[editableKind] || TextEditableStrategy
```